### ISSUE TITLE
Optionally ignore external dependencies

### ISSUE THREAD

**Issue by jamiewinder** [jrburke/r.js#748]
I want to optimize my web site which consists of some core functionality and several pages. Both the core code and page code are represented using several modules.

I want to bundle the core functionality and pages into individual files, but without them importing external dependencies. For example, if I have the following structure:
- core
  - menu.js
  - title.js
- page1
  - page.js (depends on core/menu)
  - basket.js (depends on core/title)
- page2
  - page.js (depends on core/menu)
  - submit.js (depends on core/menu, core/title)

Here, I want to end up with:
- core.js (contains core/menu, core/title)
- page1.js (contains page1/page, page1/basket)
- page2.js (contains page2/page, page2/submit)

In other words, I want to ask r.js to ignore dependencies that exist outside of a specific directory. Instead, it should assume they will be available at runtime. I don't think there is any way to do this at the moment apart from maintaining exclusions using the 'empty:' scheme in paths, but this can be laborious and easy to get wrong.
